sdk/daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV2.scala

```diff
@@ -1217,7 +1275,29 @@ private[archive] class DecodeV2(minor: LV.Minor) {
             Ret(EExperimental(experimental.getName, typ))
           }
 
-      }) { expr =>
+        case PLF.Expr.SumCase.INTERNED_EXPR =>
+          assertSince(LV.Features.exprInterning, "interned exprs unsupported in this version")
+          currentInternedExprId match {
+            case None =>
+              currentInternedExprId = Some(lfExpr.getInternedExpr)
+            case Some(currentId) if lfExpr.getInternedExpr < currentId =>
+              currentInternedExprId = Some(lfExpr.getInternedExpr)
+            case _ =>
+              throw Error.IllegalInterning(
+                "Interned expression indexes not monotonic (interned expressions may only refer to interned expressions of smaller index)"
+              )
+          }
+          decodeInternedExpr(
+            internedExprs.applyOrElse(
+              lfExpr.getInternedExpr,
+              (index: Int) => throw Error.Parsing(s"invalid internedExprs table index $index"),
+            ),
+            definition,
+          ) { expr =>
+            Ret(expr) // RB: idk how to write id here
+          }
```

Comment:
Is the question about difference between () and {}

() are used for one-line function bodies while  {} are use for multi-line function bodies, 

for instance 

```
 )(expr => 
    log("we use () here")
    Ref(expr)
)
```
will not be correct by while
) { expr => 
  log("we use {} here")
  Ret(expr)
}
is.

In your case, by convention, you should prefer the `()`, following the [Rule of least power](https://en.wikipedia.org/wiki/Rule_of_least_power).

Reply:
you have `identity` in scala.

Reply:
Maybe we could add `id` in `Work`.

```
object Work {
  ...
  def id[X](x: X): Work[X] = Ret(x) 
  ...
}
```

and 
you will write 

```suggestion 
            )
          copy(currentInternedExprId = Some(exprIdx)).decodeInternedExpr(
            internedExprs.applyOrElse(
              lfExpr.getInternedExpr,
              (index: Int) => throw Error.Parsing(s"invalid internedExprs table index $index"),
            ),
            definition,
          )(Work.id)
```

Reply:
if you do not plan to change the syntax, please remove the comment.